src/docs/sdk/unified-api/tracing.mdx

```diff
@@ -0,0 +1,59 @@
+    - Internally we can use the `Scope` to hold on to a reference of the current `Transaction` if needed
+- Introduce the new type [Span](/sdk/event-payloads/span/)
+- Introduce new type [Transaction](/sdk/event-payloads/transaction/) which inherits from `Span`
+- `Span` Interface
+    - When a `Span` is created, set the `startTimestamp` to the current time
+    - `SpanContext` is the attribute collection for a `Span` (Can be an implementation detail)
+    - The relation between parent - child is captured in the property `parentSpanId`
+- `Transaction` Interface
+    - A `Transaction` internally holds a flat list of child Spans (not a tree structure)
+    - `Transaction` has additionally a `setName` method the set the name of the transaction
+    - `Transaction` receives a `TransactionContext` on creation (new property vs. `SpanContext` is `name`)
+    - Since a `Transaction` inherits a `Span` it has all functions available and can be interacted with like it was a `Span`
+- `Hub` → Introduce a method called `startSpan`
+    - Creates a new `Span` instance
+    - If there is already a `Span` on the current `Scope`, the created `Span` should be its child
+    - This method deals with sampling, and therefore it should take the `tracesSampleRate` option into account:
+        - If it's a child `Span`, we inherit the decision from the parent
+        - If it's a root span, we need to roll the dice here
+        - Depending on the outcome, the sample decision should be stored in the `Span`'s `sampled` property and again forwarded to its children
+- `Hub` → Introduce a method called `traceHeaders`
+    - This function returns a header (string) `sentry-trace`
+    - The value should be the trace header string of the `Span` that is currently on the `Scope`
+- `Scope` Introduce `setSpan`
+    - This can be used internally to pass a `Span` / `Transaction` around so that integrations can attach children to it
+    - Setting the `transaction` property on the `Scope` (legacy) should overwrite the name of the `Transaction` stored in the `Scope`, if there is one. With that we give users the option to change the transaction name even if they don't have access to the instance of the `Transaction` directly.
+- `Span.finish()`
+    - Just set `endTimestamp` to the current time (in payload `timestamp`)
+- `Transaction.finish()`
+    - `super.finish()` (call finish on Span)
+    - Send it to Sentry only if `sampled == true`
+    - A `Transaction` needs to be wrapped in an `Envelope` and sent to the [Envelope Endpoint](https://develop.sentry.dev/sdk/envelopes/)
+    - The `Transport` should use the same internal queue for `Transactions` / `Events`
+    - The `Transport` should implement category-based rate limiting →
+    - The `Transport` should deal with wrapping a `Transaction` in an `Envelope` internally
+- A `Transaction` should **not** go through `beforeSend` (but `EventProcessors` are fine for now, since they are considered internal)
```

Comment:
I'm afraid avoiding BeforeSend and going through EventProcessor might cause confusion.
Until today we've announced: "Events always go through BeforeSend" so the user has a chance to change it. Perhaps this just shows that Transactions are not Events?

A typed EventProcessor is our public API for anyone who wants to use an instance of a type that implements an interface and doesn't care to be executed only once the event is totally formed (only BeforeSend gives that guarantee). But saying Transaction events go through those callbacks could cause confusion since they'll be "weird events" and could behave strange in users current EventProcessors.

Reply:
This is not ideal I know and maybe for now we need an internal "beforeTransaction" (we don't want to expose this either). But integrations need a way to still attach data or discard outgoing Transactions before sending. The workaround for this is to still let them go through eventProcessors but just not call the user function beforeSend.

Reply:
The problem is that we'll trigger user defined `EventProcessor` too. Users do use that in .NET which is much preferable in cases of dependency injection over an inlined function (beforeSend).
Sounds to me that a Transaction isn't an Event but I'll resolve this as clarified.